src/hotspot/os/bsd/os_perf_bsd.cpp

```diff
@@ -138,6 +141,36 @@ int CPUPerformanceInterface::CPUPerformance::cpu_load_total_process(double* cpu_
 
   *cpu_load = (double)used_delta / total_delta;
 
+  return OS_OK;
+#elif defined(__FreeBSD__)
+  // counters are user/nice/system/interrupt/idle/states
+  long cpu_load_info[6];
+  size_t length = sizeof(long) * 6;
```

Comment:
I'm reviewing from a *BSD perspective only. I'll be wanting to layer in OpenBSD support similar to how it was done in https://github.com/battleblow/openjdk-jdk11u/blob/bsd-port/src/java.base/bsd/native/libjava/ProcessHandleImpl_bsd.c#L49

Could you use the CPUSTATES define instead of hard coding the array length? From what I can tell it is available on *BSD as the length of the cp_time sysctl array.